app/features/transactions/transaction-list.tsx

```diff
@@ -207,15 +229,39 @@ export function TransactionList() {
     status,
   } = useTransactions();
 
+  const { data: notifications } = useNotifications({
```

Comment:
I don't like how here we have to fetch all notifications to join them with transactions on client side. Also what if there are hundreds or even thousands of notifications and transactions